src/dotnet/commands/dotnet-new3/New3CommandShim.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Config;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Macros;
+using Microsoft.TemplateEngine.Utils;
+
+namespace Microsoft.DotNet.Tools.New3
+{
+    internal class New3CommandShim
+    {
+        private const string HostIdentifier = "dotnetcli";
+        private const string HostVersion = "1.0.0";
+        private const string CommandName = "new3";
+
+        public static int Run(string[] args)
+        {
+            return New3Command.Run(CommandName, CreateHost(), FirstRun, args);
+        }
+
+        private static ITemplateEngineHost CreateHost()
+        {
+            var builtIns = new Dictionary<Guid, Func<Type>>
+            {
+                { new Guid("0C434DF7-E2CB-4DEE-B216-D7C58C8EB4B3"), () => typeof(RunnableProjectGenerator) },
```

Comment:
These are here to minimize the possibility that a user could end up in a state with no generators/components by establishing that these components are shipped with this host (rather than being dynamically acquired).

These map an identifier defined on the type to the actual type (without doing disk based discovery) so that a fresh initialization can figure out what's available for it to use. The code below establishes the set of preferences the CLI has that allow the user to skip making certain choices. 

"prefs:language" sets the language to use for the template in the case that there are multiple languages available for an otherwise uniquely identified template. For example, if I type `dotnet new3 mvc`, it actually matches two templates, MVC for C# and MVC for F#, this preference automatically chooses the C# flavor of the template - the user can make the choice explicitly by specifying the -lang|--language switch.

Finally, this method combines all of these bits of information, wrapping them in to a "host" object that is available to the template instantiation process so that parameter defaulting/language disambiguation/search paths/logging services/etc. can be used wherever they're needed.